cmd/server/app/server.go

```diff
@@ -142,6 +143,16 @@ func (p *Proxy) Run(o *options.ProxyRunOptions, stopCh <-chan struct{}) error {
 		defer frontendStop()
 	}
 
+	if o.LeaseLabelSelector != "" || o.LeaseDuration != 0 || o.LeaseGCInterval != 0 || o.LeaseRenewalInterval != 0 {
+		leaseLabels, err := labels.ConvertSelectorToLabelsMap(o.LeaseLabelSelector)
+		if err != nil {
+			return fmt.Errorf("could not parse lease label selector: %w", err)
```

Comment:
Having the unparseable string (using %q) in the error message makes the oncalls life better.

Reply:
Done!